src/xenia/kernel/xsocket.cc

```diff
@@ -257,5 +257,26 @@ bool XSocket::QueuePacket(uint32_t src_ip, uint16_t src_port,
   return true;
 }
 
+X_STATUS XSocket::GetSockName(uint8_t* buf, int* buf_len) {
+  struct sockaddr sa = {};
+
+  int ret = getsockname(native_handle_, &sa, (socklen_t*)buf_len);
+  if (ret < 0) {
+    return X_STATUS_UNSUCCESSFUL;
+  }
+
+  std::memcpy(buf, &sa, *buf_len);
+  return X_STATUS_SUCCESS;
+}
+
+uint32_t XSocket::GetLastWSAError() const {
+  // Todo(Gliniak): Provide error mapping table
+  // Xbox error codes might not match with what we receive from OS
+#ifdef XE_PLATFORM_WIN32
+  return WSAGetLastError();
+#endif
+  return errno;
```

Comment:
It's preferable to place this `return` in `#else` so there's no possibility of compilers warning about dead code, though I'll fix this.